src/main/java/core/mate/academy/service/ExcavatorProducer.java

```diff
@@ -0,0 +1,18 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer {
```

Comment:
The `ExcavatorProducer` class should be parameterized as `ExcavatorProducer implements MachineProducer<Excavator>` to ensure type safety and prevent the creation of invalid implementations.